From ba72d313b7dcafaf7908e50ae58888aad4c4fc93 Mon Sep 17 00:00:00 2001
From: Christian Dreher <cdreher@prophesee.ai>
Date: Tue, 18 Feb 2025 19:20:08 +0100
Subject: [PATCH 1/2] Disable packet timeout

---
 imx636.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/imx636.c b/imx636.c
index 9cf0eed..200e230 100644
--- a/imx636.c
+++ b/imx636.c
@@ -223,6 +223,7 @@ union ro_lowpower_ctrl {
 
 #define IMX636_MIPI_CONTROL (MIPI_CSI_BASE + 0x000)
 #define IMX636_MIPI_CSI_ENABLE BIT(0)
+#define IMX636_MIPI_PACKET_TIMEOUT_ENABLE BIT(5)
 
 #define IMX636_MIPI_ESCAPE_CTRL (MIPI_CSI_BASE + 0x004)
 #define IMX636_MIPI_ESCAPE_CLK_EN BIT(7)
@@ -964,6 +965,9 @@ static int imx636_reconfigure_csi2_freq(struct imx636 *imx636)
 	/* Disable MIPI CSI-2 */
 	RET_ON(imx636_clear_reg(imx636, IMX636_MIPI_CONTROL, IMX636_MIPI_CSI_ENABLE));
 
+	/* Only close packet when MIPI_PACKET_SIZE is reached */
+	RET_ON(imx636_clear_reg(imx636, IMX636_MIPI_CONTROL, IMX636_MIPI_PACKET_TIMEOUT_ENABLE));
+
 	/* Power down CSI-2 and D-PHY */
 	RET_ON(imx636_write_reg(imx636, IMX636_MIPI_STREAM, 0));
 	RET_ON(imx636_clear_reg(imx636, IMX636_MIPI_ESCAPE_CTRL, IMX636_MIPI_ESCAPE_CLK_EN));
-- 
2.39.5

